### PR TITLE
Use <mwc-fab> for the upload icon

### PR DESCRIPTION
**Pull request by beverloo**
Live on https://llaminator.peter.sh/

I'm actually not too happy about two things here, will file separate issues.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -48,7 +49,7 @@ window.addEventListener('load', () => {
     (e.target as IDBRequest).result.createObjectStore(objStoreName);
   });
 
-  fileInput.addEventListener('change', (e) => onFileInputChange(e, dbOpenRequest, imgElement, shareBtn));
+  fileInput.addEventListener('fileselected', (e) => onFileInputChange(e as CustomEvent, dbOpenRequest, imgElement, shareBtn));
```
Comment (glennhartmann) [inline src/index.ts:52]: not sure if this is valid, but I think it would be slightly cleaner like:

```
(e: CustomEvent) => onFileInputChange(e, ...)
```

**Review by glennhartmann — APPROVED** [review]

File: src/components/llama-upload-fab.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      detail: event.target.files[0],
+    }));
+  }
+
+  render() {
+    return html`
+      <input type="file" id="upload" hidden @change=${this.handleInputChange} />
+      <label id="root" for="upload">
+        <mwc-fab icon="file_upload"
+                 label="Select a file to upload to Llaminator">
```
[the remark this answers is not in the document]
Comment (beverloo) [reply src/components/llama-upload-fab.ts]: Done

File: src/components/llama-upload-fab.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      return;  // no files were selected
+
+    this.dispatchEvent(new CustomEvent('fileselected', {
+      detail: event.target.files[0],
+    }));
+  }
+
+  render() {
+    return html`
+      <input type="file" id="upload" hidden @change=${this.handleInputChange} />
```
[the remark this answers is not in the document]
Comment (beverloo) [reply src/components/llama-upload-fab.ts]: Done

File: src/index.ts (hunk 1 above)
Comment (beverloo) [reply src/index.ts:52]: There's a way, but honestly I'm not sure how. It involves typing the `fileselected` custom event, specific to the `LlamaSelectFab` element.

**Pull request merged**